measfunc/dmmqdac.py

```diff
@@ -2,182 +2,432 @@
 from qcodes.instrument.base import Instrument
 from qcodes.instrument.channel import InstrumentChannel
 from qcodes import Parameter, ParameterWithSetpoints
-from qcodes.utils.validators import Numbers, Arrays
+from qcodes.utils.validators import Numbers, Ints, Arrays
 from typing import Any, Iterable, Tuple, Union, Optional
 from time import sleep
 
+class BufferedAcquisitionController(Instrument):
+    """
+    Meant to be used with a QDac I and a Keysight digital multimeter. 
+
+    Usage:
+        # # 
+        # # Pre-init config 
+        dSETlb_dbsT = 0.0225 
+        fast_channel_setpoints = {'qdac_channel': qdac.ch01,
+            'vstart': 1.39,
+            'vend': 1.81,
+            'num_samples': 201}
+        fast_compensating_channel_setpoints = {'qdac_channel': qdac.ch02,
+            'vstart': 1.0 + dSETlb_dbsT*(fast_channel_setpoints['vend'] - fast_channel_setpoints['vstart'])/2,
+            'vend': 1.0 - dSETlb_dbsT*(fast_channel_setpoints['vend'] - fast_channel_setpoints['vstart'])/2,
+            'num_samples': 201}
+        # # 
+        # # Initialize class object 
+        trace_acquisition = BufferedAcquisitionController(name='trace_acquisition', 
+            dmm=rf_X, qdac=qdac_A, 
+            fast_channel_setpoints=fast_channel_setpoints,
+            fast_compensating_channel_setpoints=fast_compensating_channel_setpoints)
+    
+        # # 
+        # # Acquire data 
+        ydata = trace_acquisition.buffered_1d_acquisition()
+        xdata = trace_acquisition.fast_channel_setpoints.voltage_setpoints()
 
-class DMMAcquisition(Instrument):
-    def __init__(self, name: str, dmm: Instrument, qdac: Instrument,
-                 fast: InstrumentChannel,
-                 slow: InstrumentChannel,
-                 fast_n_points: int,
-                 slow_n_points: int,
-                 fast_start: float,
-                 fast_end: float,
-                 slow_start: float,
-                 slow_end: float,
-                 fast_comp: Optional[InstrumentChannel],
-                 slow_comp: Optional[InstrumentChannel],
-                 fast_comp_start: float,
-                 fast_comp_end: float,
-                 slow_comp_start: float,
-                 slow_comp_end: float,
+        plt.plot(xdata, ydata)
+
+        # # 
+        # # Change settings post-init
+        fast_compensating_channel_setpoints = {
+            'vstart': 1.0 + dSETlb_dbsT*(fast_channel_setpoints['vend'] - fast_channel_setpoints['vstart'])/2,
+            'vend': 1.0 - dSETlb_dbsT*(fast_channel_setpoints['vend'] - fast_channel_setpoints['vstart'])/2,
+            'num_samples': 201}
+        trace_acquisition.setup_channel_setpoints('fast_compensating_channel_setpoints', **fast_compensating_channel_setpoints)
+    """
+    def __init__(self, name:str, dmm:Instrument, qdac:Instrument,
+                 fast_channel_setpoints:dict, 
+                 fast_compensating_channel_setpoints:dict={},
```

Comment:
Why pass a dictionary, instead individual arguments? You can always pass a dictionary of single arguments by: **my_dict_of_arguments